src/springtime/datasets/phenocam.py

```diff
@@ -4,72 +4,206 @@
+        phenocamr = importr("phenocamr")
+        if self._exists_locally() or CONFIG.force_override:
+            logger.info(f"Phenocam files already downloaded {self._locations()}")
+        else:
+            phenocamr.download_phenocam(
+                site=self.site,
+                frequency=self.frequency,
+                internal=False,
+                out_dir=str(phenocam_data_dir),
+                **optional_args,
+            )
+
+    def _exists_locally(self) -> bool:
+        return all([location.exists() for location in self._locations()])
+
+    def _location(self, row: pd.Series) -> Path:
+        freq = "roistats"
+        if self.frequency != "roistats":
+            freq = f"{self.frequency}day"
+        return (
+            phenocam_data_dir
+            / f"{row.site}_{row.veg_type}_{row.roi_id_number:04}_{freq}.csv"
+        )
+
+    def _locations(self) -> List[Path]:
+        rois_df = list_rois()
+        rois_df = rois_df.loc[rois_df.site.str.contains(self.site)]
+        if self.veg_type is not None:
+            rois_df = rois_df.loc[rois_df.veg_type == self.veg_type]
         if self.rois is not None:
-            optional_args['roi_id'] = self.rois
-        phenocamr.download_phenocam(site=self.site, 
-                                    frequency=self.frequency,
-                                    internal=False,
-                                    out_dir=str(phenocam_data_dir),
-                                    **optional_args,
-                                    )
+            rois_df = rois_df.loc[rois_df.roi_id_number.isin(self.rois)]
 
-    def _roi_filename(self, roi: str):
-        return phenocam_data_dir / f'{self.site}_{self.veg_type}_{roi}_{self.frequency}'
+        return rois_df.apply(self._location, axis="columns")
 
     def load(self):
-        if self.rois == None:
-            rois = list_rois().loc[rois.site == self.site]
-        else:
-            rois = self.rois
-        filenames = [self._roi_filename(roi) for roi in self.rois]
+        """Load the dataset from disk into memory.
+
+        This may include pre-processing operations as specified by the context, e.g.
+        filter certain variables, remove data points with too many NaNs, reshape data.
+        """
+        df = pd.concat([_load_location(location) for location in self._locations()])
+        if self.years is not None:
+            df = df.loc[(self.years[0] <= df.year) & (df.year <= self.years[1])]
+        return df
 
 
 class PhenocamrBoundingBox(BaseModel):
+    """PhenoCam time series for sites in a bounding box.
+
+    Fetch data from https://phenocam.nau.edu/webcam/
+
+    Requires phenocamr R package.
+    Install with
+    ```R
+    install.packages("phenocamr")
+    ```
+    """
+
     dataset: Literal["phenocambbox"] = "phenocambbox"
-    area: NamedArea
+    box: Tuple[float, float, float, float]
+    """Bounding box as top left / bottom right pair (maxlat,minlon,minlat,maxlon)."""
```

Comment:
This way of defining bbox is a bit strange and different from the other datasets. Maybe it is better to be: (minlon, minlat,maxlon,maxlat)

Reply:
Let's address that in #92